### ISSUE TITLE
Fixing role dependency chain creation

### ISSUE THREAD

**Issue by basictheprogram** [ansible/ansible#14099]
##### Issue Type:
- Bug Report
##### Ansible Version:

$ ansible --version
ansible 2.1.0 (detached HEAD fb797a9e77) last updated 2016/01/25 02:29:33 (GMT -500)
  lib/ansible/modules/core: (detached HEAD 09e2457eb0) last updated 2016/01/22 08:24:22 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD e8427cb32a) last updated 2016/01/22 08:24:22 (GMT -500)
  config file = /Users/tanner/projects/ansible.git/playbooks.git/ansible.cfg
  configured module search path = Default w/o overrides

Multiple plays worked before fb797a9e7786a3fde70e7de34ccf9ac3946501b3 and now are broken. Doing a git bisect I found this is the commit that breaks things.

commit fb797a9e7786a3fde70e7de34ccf9ac3946501b3
Author: James Cammarata jimi@sngx.net
Date:   Fri Jan 22 12:44:56 2016 -0500

```
Fixing role dependency chain creation

The dep chain for roles created during the compile step had bugs, in
which the dep chain was overwriten and the original tasks in the role
were not assigned a dep chain. This lead to problems in determining
whether roles had already run when in a "diamond" structure, and in
some cases roles were not correctly getting variables from parents.

Fixes #14046
```

```

---
# file: virtual-servers.yml

- hosts: all
  roles:
    - { role: virtual-servers, become: yes }
  tasks:
```

```

---
# tasks file for virtual-servers tasks/main.yml

- include: debian.yml
  when: ansible_os_family == 'Debian'
- include: redhat.yml
  when: ansible_os_family == 'RedHat'
```

```

---
# tasks file for virtual-servers tasks/debian.yml

- name: debugging
  debug: msg="debugging"
  tags: virtual
```

_BEFORE_ the fb797a9e7786a3fde70e7de34ccf9ac3946501b3 commit

```
$ ansible-playbook -i customers/testing --ask-vault-pass --user=testing virtual-servers.yml
Vault password:

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [downloads.testing.com]

TASK [virtual-servers : include] ***********************************************
included: /Users/tanner/projects/ansible.git/playbooks.git/roles/virtual-servers/tasks/debian.yml for downloads.testing.com

TASK [virtual-servers : debugging] *********************************************
ok: [downloads.testing.com] => {
    "msg": "debugging"
}
<snip>
```

_AFTER_ applying the fb797a9e7786a3fde70e7de34ccf9ac3946501b3 commit

```
Vault password:

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [downloads.testing.com]

TASK [virtual-servers : include] ***********************************************
included: /Users/tanner/projects/ansible.git/playbooks.git/roles/virtual-servers/tasks/debian.yml for downloads.testing.com

PLAY RECAP *********************************************************************
downloads.testing.com : ok=2    changed=0    unreachable=0    failed=0
```

**Comment by zarnovican** [comment]
Can you please re-test? I also found severe problem (https://github.com/ansible/ansible/issues/14046#issuecomment-174446135) with that commit, which was fixed by subsequent a8ffa021344b5548813790f18dda68d32717a4d7 commit.

**Comment by basictheprogram** [comment]
The below ansible version resolves the problem (which includes your fix). Thank you!

Closing this ticket.

$ ansible --version
ansible 2.1.0 (devel 9abb107044) last updated 2016/01/25 13:07:24 (GMT -500)
  lib/ansible/modules/core: (detached HEAD 09e2457eb0) last updated 2016/01/22 08:24:22 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD e8427cb32a) last updated 2016/01/22 08:24:22 (GMT -500)
  config file = /Users/tanner/projects/ansible.git/playbooks.git/ansible.cfg
  configured module search path = Default w/o overrides
